### ISSUE TITLE
Performance on the released model

### ISSUE THREAD

**Issue by LrHITer** [Delong-liu-bupt/Word4Per#11]
Great work!
When I was using the released pre-trained model for inference, I found that the results were significantly different from those in paper. May I ask if I am missing any important operations? Thanks!

My training_hyperparameters.json is:
```
{
    "backbone": "pretrain",
    "batch_size": 256,
    "blip_model_name": "blip2_fafa_cpr",
    "exp_name": "FAFA_SynCPR_FDA_FD_MFR",
    "fd_margin": 0.5,
    "fda_alpha": 0.5,
    "fda_k": 6,
    "itcpr_root": "/data/FAFA_SynCPR/ITCPR",
    "json_path": "SynCPR.json",
    "learning_rate": 2e-06,
    "loss_fd": 1.0,
    "loss_fda": 1.0,
    "loss_mfr": 0.5,
    "num_epochs": 10,
    "num_workers": 4,
    "resume_path": null,
    "save_best": true,
    "save_last": false,
    "save_memory": false,
    "save_training": true,
    "setting": [
        "annotations"
    ],
    "syncpr_data_path": "/data/FAFA_SynCPR/SynCPR/",
    "target_ratio": 1.25,
    "transform": "squarepad",
    "validation_frequency": 1,
    "validation_step": 500
}
```
The inference result through the released pre-trained model is:
<img width="650" height="798" alt="Image" src="https://github.com/user-attachments/assets/3e428c58-8c47-4ae5-b00b-983356f0be9f" />

**Comment by Delong-liu-bupt** [comment]
Thanks for your interest in our work and for providing your configuration details.

Regarding the parameters you posted, they look correct for inference. To investigate this potential issue, I just re-downloaded the open-sourced model and ran the inference using the exact configuration you provided.

However, I was unable to reproduce the discrepancy. I noticed from your previous screenshot that the number of samples in your test set appears correct, so the data loading seems to be working fine. Since the model file and parameters function correctly on my side, I am not entirely sure what might be causing the low performance in your environment.

<img width="1159" height="660" alt="Image" src="https://github.com/user-attachments/assets/80fb3c95-0f1d-42a7-94a0-ee8edad46f64" />

**Comment by LrHITer** [comment]
Thank you very much for your reply!

My environment: completely configured according to the Setup you provided.

My dataset: Since the processing of Celeb-reID in your img_process.py does not include "train" part, but the images in "train" part are included in gallery.json and query.json, I added "train" part in img_process.py.
`subfolders = ['train', 'gallery', 'query']  # List of subfolders`

My model weights: I manually downloaded some corresponding model weights, including bert-base-uncased, eva_vit_g.pth, and blip2_pretained.pth, and modified the corresponding paths in blip2.py and eva-vit.py.
```
https://hf-mirror.com/google-bert/bert-base-uncased
https://storage.googleapis.com/sfr-vision-language-research/LAVIS/models/BLIP2/blip2_pretrained.pth
https://storage.googleapis.com/sfr-vision-language-research/LAVIS/models/BLIP2/eva_vit_g.pth
```

My training_hyperparameters.json: I did not complete the model training process, I only ran the training command to obtain the training_hyperparameters.json file.

I have tried the inference process on different machines, including 3090, 4090, and a100, and the results obtained are only slightly different in mAP and mINP. 

So I'm not sure if you can generously provide possible operations that can still be tried?

**Comment by Delong-liu-bupt** [comment]
Hi, thanks for the detailed update regarding your environment and modifications.

Regarding your setup and data: Your modification to img_process.py (adding the "train" part) seems reasonable to ensure the dataset structure aligns with the JSON files. Generally speaking, if there were a critical error in data preprocessing or loading, the inference code would likely crash or throw an exception rather than completing with low metrics. Since the code runs through, the data pipeline is likely functioning correctly.

Regarding the config: Generating the training_hyperparameters.json by briefly running the training command is also a valid approach. The parameters should be compatible. Just to be absolutely sure, you can verify your config against the one I used. I have uploaded my original configuration file here: https://huggingface.co/datasets/a1557811266/SynCPR/blob/main/training_hyperparameters.json

Suggestions: I find it difficult to pinpoint a specific flaw in your workflow, as your operations on different machines (3090, 4090, A100) yield consistent results.

Directory Structure: Please ensure you have created the correct experiment directory and placed the downloaded checkpoint file inside it exactly as required.

File Integrity: Since other researchers have successfully reproduced the test performance (as noted in previously closed issues), the code and model should be correct. If you still cannot get the expected results, I strongly suggest checking the integrity (MD5/SHA) of the manually downloaded weights (bert, eva_vit, blip2, tuned_recall_at1_step.pt) and verifying that the test set images are not corrupted during download or extraction.

Hope this helps!
